File: mcp/ts/src/tools/consolidate.ts
// SPDX-License-Identifier: Apache-2.0

import { z } from 'zod'
import { jsonContent, type Tool } from './types.js'

const schema = z.object({
  brain: z.string().optional(),
})

export const consolidateTool: Tool<typeof schema> = {
  name: 'memory_consolidate',
  description:
    'Trigger a consolidation pass on the brain (compile summaries, promote stable notes, prune stale episodic memory).',
  inputSchema: schema,
  async handler(args, client, ctx) {
    const result = await client.consolidate(args, ctx?.progress)
    return jsonContent(result)
  },
}

File: go/cmd/memory-mcp/tools/consolidate.go
// SPDX-License-Identifier: Apache-2.0

package tools

import (
	"context"

	"github.com/google/jsonschema-go/jsonschema"
	"github.com/modelcontextprotocol/go-sdk/mcp"
)

func registerConsolidate(server *mcp.Server, client MemoryClient) {
	schema := &jsonschema.Schema{
		Type: "object",
		Properties: map[string]*jsonschema.Schema{
			"brain": {Type: "string"},
		},
	}
	mcp.AddTool(server, &mcp.Tool{
		Name:        "memory_consolidate",
		Description: "Trigger a consolidation pass on the brain (compile summaries, promote stable notes, prune stale episodic memory).",
		InputSchema: schema,
	}, func(ctx context.Context, req *mcp.CallToolRequest, args ConsolidateArgs) (*mcp.CallToolResult, any, error) {
		progress := progressFromRequest(ctx, req)
		result, err := client.Consolidate(ctx, args, progress)
		if err != nil {
			return nil, nil, err
		}
		return structuredResult(result)
	})
}
